plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -0,0 +1,422 @@
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import org.apache.log4j.Logger;
+
+import azkaban.jobExecutor.JavaProcessJob;
+import azkaban.security.commons.HadoopSecurityManager;
+import azkaban.security.commons.HadoopSecurityManagerException;
+import azkaban.utils.Props;
+import azkaban.utils.StringUtils;
+
+public class HadoopSparkJob extends JavaProcessJob {
+
+  // Azkaban/Java params
+  private static final String USER_TO_PROXY = "user.to.proxy";
+
+  private static final String HADOOP_SECURE_SPARK_WRAPPER = HadoopSecureSparkWrapper.class
+          .getName();
+
+  private static final Logger logger = Logger.getRootLogger();
+
+  // Spark params
+  private static final String MASTER = "master";
+
+  private static final String JARS_PREFIX = "jars";
+
+  private static final String EXECUTION_JAR_AZKABAN = "execution.jar";
+
+  private static final String EXECUTION_CLASS_AZKABAN = "execution.class";
+
+  private static final String EXECUTION_CLASS_SPARK = "class";
+
+  private static final String NUM_EXECUTORS_AZKABAN = "num.executors";
+
+  private static final String NUM_EXECUTORS_SPARK = "num-executors";
+
+  private static final String EXECUTOR_CORES_AZKABAN = "executor.cores";
+
+  private static final String EXECUTOR_CORES_SPARK = "executor-cores";
+
+  private static final String QUEUE = "queue";
+
+  private static final String DRIVER_MEMORY_AZKABAN = "driver.memory";
+
+  private static final String DRIVER_MEMORY_SPARK = "driver-memory";
+
+  private static final String EXECUTOR_MEMORY_AZKABAN = "executor.memory";
+
+  private static final String EXECUTOR_MEMORY_SPARK = "executor-memory";
+
+  // Spark default
+  private static final String MASTER_DEFAULT = "yarn-cluster";
+
+  private static final int NUM_EXECUTORS_DEFAULT = 2;
+
+  private static final int EXECUTOR_CORES_DEFAULT = 1;
+
+  private static final String QUEUE_DEFAULT = "marathon";
+
+  private static final String DRIVER_MEMORY_DEFAULT = "2g";
+
+  private static final String EXECUTOR_MEMORY_DEFAULT = "1g";
+
+  // security variables
+  private String userToProxy = null;
+
+  private boolean shouldProxy = false;
+
+  private boolean obtainTokens = false;
+
+  private HadoopSecurityManager hadoopSecurityManager;
+
+  private static final String HADOOP_SECURITY_MANAGER_CLASS_PARAM = "hadoop.security.manager.class";
+
+  // FileNameFilter for only jar files
+  private static FilenameFilter jarFilter = new FilenameFilter() {
+    @Override
+    public boolean accept(File dir, String name) {
+      if (name.endsWith("\\.jar"))
+        return true;
+      else
+        return false;
+    }
+  };
+
+  public HadoopSparkJob(String jobid, Props sysProps, Props jobProps, Logger log) {
+    super(jobid, sysProps, jobProps, log);
+
+    getJobProps().put("azkaban.job.id", jobid);
+
+    shouldProxy = getSysProps().getBoolean("azkaban.should.proxy", false);
```

Comment:
Let's make this a constant since it is being used more than once - "azkaban.should.proxy"